hooks/pre_gen_project.py

```diff
@@ -3,16 +3,69 @@
 Validate variables.
 """
 import re
+import subprocess  # nosec
 import sys
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import List, Sequence, Union
+
+
+def run_cmd(cmd: "Union[str, Sequence[str]]") -> subprocess.CompletedProcess:
+    """Run a command via `subprocess.run()`.
+
+    Parameters:
+        cmd: The command to run as a subprocess. If this is a string, it will be
+            converted to a list via the `split()`-method.
+
+    Returns:
+        The complete process object from the `subprocess` run.
+
+    """
+    cmd = cmd.split() if isinstance(cmd, str) else list(cmd)
+    try:
+        result_run = subprocess.run(cmd, check=True, capture_output=True)  # nosec
+    except subprocess.CalledProcessError as exc:
+        sys.exit(
+            f"ERROR: Running cmd {cmd!r} in generated project dir "
+            f"{Path.cwd().resolve()} failed.\n\nException: {exc}"
+        )
+    return result_run
 
 
 # Ensure package_name is a valid Python module name
-MODULE_REGEX = r"^[_a-zA-Z][_a-zA-Z0-9]+$"
-module_name = "{{ cookiecutter.package_name }}"
-if not re.match(MODULE_REGEX, module_name):
-    sys.exit(f"ERROR: {module_name!r} is not a valid Python module name!")
+MODULE_REGEX = re.compile(r"^[_a-zA-Z][_a-zA-Z0-9]+$")
+MODULE_NAME = "{{ cookiecutter.package_name }}"
+if not MODULE_REGEX.match(MODULE_NAME):
+    sys.exit(f"ERROR: {MODULE_NAME!r} is not a valid Python module name!")
 
 
 # Ensure package_name starts with `"oteapi_"`
-if not module_name.startswith("oteapi_"):
-    sys.exit(f"ERROR: package_name ({module_name!r}) must start with 'oteapi_'.")
+if not MODULE_NAME.startswith("oteapi_"):
```

Comment:
This requirement for package_name is not included in the readme. I think we should add it to the readme or consider removing this requirement.

Reply:
Nice catch. It should be removed completely.

Reply:
Hmm. Thinking more about this, as this is from the generated template repository, I think we should try to push people towards naming their packages `oteapi-<something` (and hence the module name `oteapi_<something>`).
But an actual failure should not occur - I agree. I think instead we can print a warning or something?

Reply:
If we want to push people towards naming their packages like that I think we should mention it in the readme (e.g. in the table). I also think that printing a warning is a good idea.

Reply:
A warning is now printed.
I'll add a commit updating the README as well 👍